src/Shared/Recruit.Vacancies.Client/Infrastructure/EventHandlers/VacancyClosedHandler.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Esfa.Recruit.Vacancies.Client.Domain.Events;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.QueryStore;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Esfa.Recruit.Vacancies.Client.Infrastructure.EventHandlers
+{
+    public class VacancyClosedHandler : INotificationHandler<VacancyClosedEvent>
+    {
+        private readonly ILogger<VacancyClosedHandler> _logger;
+        private readonly IQueryStoreWriter _queryStore;
+
+        public VacancyClosedHandler(ILogger<VacancyClosedHandler> logger, IQueryStoreWriter queryStore)
+        {
+            _logger = logger;
+            _queryStore = queryStore;
+        }
+
+        public async Task Handle(VacancyClosedEvent notification, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"Deleting LiveVacancy {notification.VacancyReference} from query store.");
+            await _queryStore.DeleteLiveVacancyAsync(notification.VacancyReference);
```

Comment:
There will be a period where the vacancy is in the search index but not available for viewing.

Reply:
I think there will be another story where we will remove the vacancy from the index but it wasn't as part of this ticket.